libc/include/llvm-libc-macros/math-function-macros.h

```diff
@@ -11,6 +11,19 @@
 
 #include "math-macros.h"
 
+#ifndef __cplusplus
+#define issignaling(x)                                                         \
+  _Generic((x),                                                                \
+      float: issignalingf,                                                     \
+      double: issignaling,                                                     \
+      long double: issignalingl)(x)
+#define iscanonical(x)                                                         \
+  _Generic((x),                                                                \
+      float: iscanonicalf,                                                     \
+      double: iscanonical,                                                     \
+      long double: iscanonicall)(x)
+#endif
```

Comment:
What happens for `__cplusplus`? We don't have `issignaling` or `iscanonical`?

Reply:
This macro is defined in C23 and while C++ does not have anything equivalent, it will most likely be `std::issignaling` as a template function. Also, the C++ header cmath in libc++ will most likely undef all the macro defs. So the tests in include should only test the function under C for now.

Reply:
However, we can keep the overloading set of functions in C++ mode, @lntue and I discussed about it and wanted to check with you about it.

Reply:
SGTM